### PR TITLE
Use RunPeriod names in error messages and use right array for SizingPeriod:WeatherFileXXX

### LINKED ISSUE #6937: RunPeriod names in error messages
Issue overview
--------------
Many of the error messages for RunPeriod identify the object as `RunPeriod object #1` instead of using the RunPeriod name:
```
   ** Warning ** RunPeriod: object #1, start weekday (THURSDAY) does not match the start year (2019), corrected to TUESDAY.
```
Requiring that the RunPeriods have names would remove the need for filling in a name and would make it easier to identify the object that has the issue.

### Details
Some additional details for this issue (if relevant):
 - Platform: All
 - Version of EnergyPlus: 8.9+

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created: #7068

### LINKED ISSUE #6949: Fans table End Use header incorrect
Issue overview
--------------
The `Equipment Summary` > `Fans` table `End Use` header is incorrect, it should be `End Use Subcategory` [here](https://github.com/NREL/EnergyPlus/blob/75a40dfc84745b3be24f5f782ed0a0b0eb126d5f/src/EnergyPlus/OutputReportPredefined.cc#L1877).

### Details
Some additional details for this issue (if relevant):
 - Platform (Windows 10)
 - Version of EnergyPlus (8.9)
 - Unmethours link or helpdesk ticket number (NA)

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### LINKED ISSUE #7067: GetRunPeriodDesignData issues warning with reference to the wrong Array
Issue overview
--------------

Found while working on #6937

Here's an example: https://github.com/jmarrec/EnergyPlus/blob/a7c9cc14ce05cf09dca6623de97faa8546e2e1bb/src/EnergyPlus/WeatherManager.cc#L6435:L6436

Notice that it uses correctly `RunPeriodDesignInput(Count).title`, but also incorrectly `RunPeriodInput(Loop).startDay`. This means that unless you have the same exact count of RunPeriod versus SizingPeriod:WeatherFileDays + SizingPeriod:WeatherFileConditions, it'll throw.

Will add a unit test for it and fix it.

### Details
Some additional details for this issue (if relevant):
 - Platform: All
 - Version of EnergyPlus: current develop a7c9cc14ce05cf09dca6623de97faa8546e2e1bb

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

* Fix #6937: I didn't do any IDD change to make the name of the `RunPeriod` a required field. By default the InputProcessor will attribute a name such as "RUN PERIOD 1" when it's empty.
Regardless or whether the name (in the code `title`) is autogenerated or not, use that in the Warning/Severe messages.
* Fix #7067: 
    * In GetRunPeriodDesignData, some warning messages were referring to the wrong array, which could end up throwing an assert contains.
     * Also, 4 warnings messages referred to the object number rather than its name like in #6937, so I fixed that for consistency. But there's 0 chance it'll ever reach them: the code checks if Yes/No fields are appropriate, and the InputProcessor would fatal before it reaches them. That's also why I didn't add an explicit unit test for this portion (can't do).

Bonus: Fix #6949: really trivial so not worth adding it to another PR

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: src/EnergyPlus/OutputReportPredefined.cc [hunk 1]
```diff
@@ -1874,7 +1874,7 @@ namespace OutputReportPredefined {
         pdchFanPwrPerFlow = newPreDefColumn(pdstFan, "Rated Power Per Max Air Flow Rate [W-s/m3]");
         pdchFanMotorIn = newPreDefColumn(pdstFan, "Motor Heat In Air Fraction");
         pdchFanEnergyIndex = newPreDefColumn(pdstFan, "Fan Energy Index");
-        pdchFanEndUse = newPreDefColumn(pdstFan, "End Use");
+        pdchFanEndUse = newPreDefColumn(pdstFan, "End Use Subcategory");
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReportPredefined.cc:1877]: This is a good change.

File: src/EnergyPlus/WeatherManager.cc [hunk 2]
```diff
@@ -6051,6 +6051,8 @@ namespace WeatherManager {
 
             ++Count;
             // Loop = RP + Ptr;
+            // Note JM 2018-11-20: IDD allows blank name, but input processor will create a name such as "RUNPERIOD 1" anyways
+            // which is fine for our reporting below
```
Comment (Myoldmopar) [inline src/EnergyPlus/WeatherManager.cc:6055]: I don't really like the date being in here, but it's not enough to hold up the whole PR if nothing else needs changing.

File: src/EnergyPlus/WeatherManager.cc [hunk 3]
```diff
@@ -6307,7 +6309,7 @@ namespace WeatherManager {
             } else if (UtilityRoutines::SameString(cAlphaArgs(8), "YES")) {
                 RunPeriodInput(Loop).actualWeather = true;
             } else {
-                ShowSevereError(cCurrentModuleObject + ": object #" + TrimSigDigits(Loop) + cAlphaFieldNames(8) + " invalid [" + cAlphaArgs(8) + ']');
+                ShowSevereError(cCurrentModuleObject + ": object=" + RunPeriodInput(Loop).title + cAlphaFieldNames(8) + " invalid [" + cAlphaArgs(8) + ']');
```
Comment (Myoldmopar) [inline src/EnergyPlus/WeatherManager.cc:6312]: This is a good change!

File: tst/EnergyPlus/unit/RunPeriod.unit.cc [hunk 4]
```diff
@@ -276,3 +276,159 @@ TEST_F(EnergyPlusFixture, RunPeriod_EndYearOnly)
...
+        // This should just issue a severe
+        EXPECT_TRUE(ErrorsFound);
+
+        std::string const error_string =
+            delimited_string({"   ** Severe  ** SizingPeriod:WeatherFileDays: object=WEATHER FILE SIZING PERIOD Begin Day of Month invalid (Day of Month) [31]"});
+
+        EXPECT_TRUE(compare_err_stream(error_string, true));
+
+    }
+}
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/RunPeriod.unit.cc:434]: Tests look like they cover your fixes.  :+1:

**Review by Myoldmopar — APPROVED** [review]
Code changes look good.  I'm going to pull it locally and merge develop into it.  I'll run tests and play with it a little and if all is well I'll get it merged in.

**Comment by jasondegraw** [comment]
Does this still not require that RunPeriod objects have names? If not, I do not agree that it is a fix for my issue.

**Comment by Myoldmopar** [comment]
Good call, error messages look nicer now, but your issue is different than what this addresses.

**Comment by Myoldmopar** [comment]
Local build was successful, unit test suite passes, ran a sampling of integration tests, everything is great.  I'm going to go ahead and merge this in.  I'll close #6949 and #7067 as complete and comment on #6937 saying it was partially addressed.  We should have a separate branch to fix #6937 as it will take transition rules.

**Pull request merged**